### PR TITLE
[DPE-6161] Document terminology decisions

### PR DESCRIPTION
**Pull request by izmalk**
Add the Terminology section to the CONTRIBUTING.md file.
Fix terminology in Readme files, CONTRIBUTING.md, and yaml specs.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -69,6 +69,36 @@ tox run -e integration   # integration tests
...
+| Apache Kafka cluster | | Charmed Apache Kafka cluster |
+
+The full form must be used at least once per page.
+The full form must be used at the first entry to the page’s headings, body of text, callouts, and graphics.
+For subsequent usage, the full form can be substituted by alternatives.
+
 ## Canonical Contributor Agreement
 
-Canonical welcomes contributions to the Charmed Kafka Operator. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
+Canonical welcomes contributions to the Charmed Apache Kafka Operator. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
```
Comment (deusebio) [inline CONTRIBUTING.md:104]:
```suggestion
Canonical welcomes contributions to Charmed Apache Kafka. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
```

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ The Charmed Operator can be found on [Charmhub](https://charmhub.io/kafka) and i
 - SASL/SCRAM auth for Broker-Broker and Client-Broker authentication enabled by default.
 - Access control management supported with user-provided ACL lists.
 
-As currently Kafka requires a paired ZooKeeper deployment in production, this operator makes use of the [ZooKeeper Operator](https://github.com/canonical/zookeeper-operator) for various essential functions.
+As currently Apache Kafka requires a paired Apache ZooKeeper deployment in production, this operator makes use of the [Apache ZooKeeper Operator](https://github.com/canonical/zookeeper-operator) for various essential functions.
```
Comment (deusebio) [inline README.md:17]:
```suggestion
As currently Apache Kafka requires a paired Apache ZooKeeper deployment in production, this operator makes use of [Charmed Apache ZooKeeper](https://github.com/canonical/zookeeper-operator) for various essential functions.
```

File: README.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ For more information on how to perform typical tasks, see the How to guides sect
 
 ### Deployment
 
-The Kafka and ZooKeeper operators can both be deployed as follows:
+The Apache Kafka and Apache ZooKeeper operators can both be deployed as follows:
```
Comment (deusebio) [inline README.md:54]:
```suggestion
Charmed Apache Kafka and Charmed Apache ZooKeeper can both be deployed as follows:
```

Comment (deusebio) [reply README.md:54]: if you really want to use operator, i have noticed that elsewhere you were capitalizing that. Anyhow I would just drop operators (unless we want to only refer to the charm code), but here for instance we refer to both operator code and the workload, so I'd say that we deploy Charmed Apache Kafka and Charmed Apache ZooKeeper

File: README.md [hunk 4]
```diff
@@ -70,18 +70,18 @@ To watch the process, the `juju status` command can be used. Once all the units
 juju run-action kafka/leader get-admin-credentials --wait
 ```
 
-Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, e.g `bin/kafka-config.sh` to update cluster configuration, `bin/kafka-topics.sh` for topic management, and many more! The Kafka Charmed Operator provides these commands for administrators to run their desired cluster configurations securely with SASL authentication, either from within the cluster or as an external client.
+Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, e.g `bin/kafka-config.sh` to update cluster configuration, `bin/kafka-topics.sh` for topic management, and many more! The Apache Kafka Charmed Operator provides these commands for administrators to run their desired cluster configurations securely with SASL authentication, either from within the cluster or as an external client.
```
Comment (deusebio) [inline README.md:73]:
```suggestion
Apache Kafka ships with `bin/*.sh` commands to do various administrative tasks, e.g `bin/kafka-config.sh` to update cluster configuration, `bin/kafka-topics.sh` for topic management, and many more! Charmed Apache Kafka provides these commands for administrators to run their desired cluster configurations securely with SASL authentication, either from within the cluster or as an external client.
```

Apache Kafka Charmed Operator sounds weird. Also, it is actually the snap that provides these commands, not the charm code. Anyhow, it is a small difference, I'd just use Charmed Apache Kafka

File: README.md [hunk 5]
```diff
@@ -70,18 +70,18 @@ To watch the process, the `juju status` command can be used. Once all the units
...
+For example, to list the current topics on the Apache Kafka cluster, run the following command:
 
 ```shell
 BOOTSTRAP_SERVERS=$(juju run-action kafka/leader get-admin-credentials --wait | grep "bootstrap.servers" | cut -d "=" -f 2)
 juju ssh kafka/leader 'charmed-kafka.topics --bootstrap-server $BOOTSTRAP_SERVERS --list --command-config /var/snap/charmed-kafka/common/client.properties'
 ```
 
-Note that Charmed Apache Kafka cluster is secure-by-default: when no other application is related to Kafka, listeners are disabled, thus preventing any incoming connection. However, even for running the commands above, listeners must be enabled. If there are no other applications, you can deploy a `data-integrator` charm and relate it to Kafka to enable listeners.
+Note that Charmed Apache Kafka cluster is secure-by-default: when no other application is related to Apache Kafka, listeners are disabled, thus preventing any incoming connection. However, even for running the commands above, listeners must be enabled. If there are no other applications, you can deploy a `data-integrator` charm and relate it to Charmed Apache Kafka to enable listeners.
```
Comment (deusebio) [inline README.md:82]:
```suggestion
Note that Charmed Apache Kafka cluster is secure-by-default: when no other application is related to Charmed Apache Kafka, listeners are disabled, thus preventing any incoming connection. However, even for running the commands above, listeners must be enabled. If there are no other applications, you can deploy a `data-integrator` charm and relate it to Charmed Apache Kafka to enable listeners.
```

File: README.md [hunk 6]
```diff
@@ -70,18 +70,18 @@ To watch the process, the `juju status` command can be used. Once all the units
...
 
-Available Kafka bin commands can be found with:
+Available Apache Kafka bin commands can be found with:
```
Comment (deusebio) [inline README.md:84]:
```suggestion
Available Charmed Apache Kafka bin commands can be found with:
```

These are available commands in the snap (so the solution), not Apache Kafka (which may even have more or different commands)

File: metadata.yaml [hunk 7]
```diff
@@ -3,12 +3,12 @@
...
 description: |
-  Kafka is an event streaming platform. This charm deploys and operates Kafka on
+  Apache Kafka is an event streaming platform. This charm deploys and operates Apache Kafka on
   a VM machines environment.
 
   Apache Kafka is a free, open source software project by the Apache Software Foundation. 
-  Users can find out more at the [Kafka project page](https://kafka.apache.org/).
-summary: Charmed Kafka Operator
+  Users can find out more at the [Apache Kafka project page](https://kafka.apache.org/).
+summary: Charmed Apache Kafka Operator
```
Comment (deusebio) [inline metadata.yaml:11]: `Operator` or `operator`?

**Review by deusebio — APPROVED** [review]
Some nipticks, nothing blocking

File: README.md (hunk 3 above)
Comment (izmalk) [reply README.md:54]: It's not that I specifically wanted to use that term. It was here long before me and out of scope for this PR.  But you're right, it does feel much better with the solution context instead. Good catch!

File: metadata.yaml (hunk 7 above)
Comment (izmalk) [reply metadata.yaml:11]: Ideally, it should be a lower-case operator.
```suggestion
summary: Charmed Apache Kafka operator
```

**Review by marcoppenheimer — CHANGES_REQUESTED** [review]
Same comments as - https://github.com/canonical/kafka-k8s-operator/pull/158#pullrequestreview-2509257999

File: README.md [hunk 8]
```diff
@@ -270,4 +270,4 @@ Also, if you truly enjoy working on open-source projects like this one, check ou
 
 ## License
 
-The Charmed Apache Kafka Operator is free software, distributed under the Apache Software License, version 2.0. See [LICENSE](https://github.com/canonical/kafka-operator/blob/main/LICENSE) for more information.
+Charmed Apache Kafka is a free software, distributed under the Apache Software License, version 2.0. See [LICENSE](https://github.com/canonical/kafka-operator/blob/main/LICENSE) for more information.
```
Comment (izmalk) [inline README.md:273]:
```suggestion
Charmed Apache Kafka is free software, distributed under the Apache Software License, version 2.0. See [LICENSE](https://github.com/canonical/kafka-operator/blob/main/LICENSE) for more information.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -69,6 +69,36 @@ tox run -e integration   # integration tests
...
+Apache®, [Apache Kafka, Kafka®](https://kafka.apache.org/), [Apache ZooKeeper, ZooKeeper™](https://zookeeper.apache.org/) and their respective logos are either registered trademarks or trademarks of the [Apache Software Foundation](https://www.apache.org/) in the United States and/or other countries.
+
+For documentation in this repository the following conventions are applied (see the table below).
+
+| Full form | Alternatives | Incorrect examples |
+| -------- | ------- | ------- |
+| Apache Kafka | | Kafka |
+| Charmed Apache Kafka | | Charmed Kafka |
+| Kafka Connect | | Kafka connect |
+| Kafka Brokers | | Kafka brokers|
```
Comment (izmalk) [inline CONTRIBUTING.md:95]:
```suggestion
| Apache Kafka brokers | | Kafka brokers, Apache Kafka Brokers |
```

**Review by marcoppenheimer — APPROVED** [review]

**Pull request merged**